### ISSUE TITLE
DL3002 has false positives in multistage builds

### ISSUE THREAD

**Issue by ianfixes** [hadolint/hadolint#405]
- [x] This is a bug report
- [x] I searched existing issues before opening this one

### Expected behavior
Only the last stage in a multistage build is considered for `DL3002` rule violations

### Actual behavior
Every stage in a multistage build seems to trigger a `DL3002` rule violation

### Steps to reproduce the behavior
```console
$ docker run --rm -i hadolint/hadolint hadolint -v
Haskell Dockerfile Linter v1.17.5-6-gbc8bab9
```

`DL3002.Dockerfile`:
```Dockerfile
FROM alpine:3.7 as stage
USER root

FROM alpine:3.7
USER operator
```

```console
$ docker run --rm -i hadolint/hadolint hadolint - < DL3002.Dockerfile 
/dev/stdin:2 DL3002 Last USER should not be root
```

**Comment by zemanlx** [comment]
Thank you @ianfixes for reporting this issue. Unfortunately, this issue can be expanded in a way that it is no longer false positive and that depends on how you use your Dockerfile.

Look at our [Dockerfile](https://github.com/hadolint/hadolint/blob/master/docker/Dockerfile) which ends with

```Dockerfile
FROM debian:stretch-slim AS debian-distro
COPY --from=builder /root/.local/bin/hadolint /bin/
CMD ["/bin/hadolint", "-"]

FROM scratch AS distro
COPY --from=builder /root/.local/bin/hadolint /bin/
CMD ["/bin/hadolint", "-"]

FROM alpine:3 AS alpine-distro
COPY --from=builder /root/.local/bin/hadolint /bin/
CMD ["/bin/hadolint", "-"] 
```

If we were using `USER root` in the first two, but not in the last one it would be an issue. During the static analysis, we do not know which images are there just for builds and which ones are for distribution.

For your use case, I think it would be best to use [Inline ignores](https://github.com/hadolint/hadolint#inline-ignores).

**Comment by czepiec** [comment]
I agree with @zemanlx's example, but I believe the vast majority of multi-stage builds use only the last container, with the others running as `root` due to the build process. Would you consider adding a configuration parameter to evaluate `DL3002` only in the last stage?
